subprojects/build-cache/src/integTest/groovy/org/gradle/caching/local/internal/DirectoryBuildCacheCleanupIntegrationTest.groovy

```diff
@@ -55,370 +54,157 @@ class DirectoryBuildCacheCleanupIntegrationTest extends AbstractIntegrationSpec
-        def oldestCacheEntry = TimeUnit.DAYS.toMillis(newList.size())
-        ageOfCacheEntries.every { it < oldestCacheEntry }
-
-        // Cache should be under the target size
-        calculateCacheSize(newList) <= MAX_CACHE_SIZE
-    }
-
-    def "cleans up based on LRU"() {
-        when:
-        runMultiple(MAX_CACHE_SIZE*2)
-        then:
-        def originalList = listCacheFiles()
-        // build cache hasn't been cleaned yet
-        originalList.size() == MAX_CACHE_SIZE*2
-        calculateCacheSize(originalList) >= MAX_CACHE_SIZE
-
-        when:
-        def oldTime = System.currentTimeMillis() - TimeUnit.DAYS.toMillis(100)
-        originalList.each { cacheEntry ->
-            cacheEntry.lastModified = oldTime
-        }
-        and:
-        withBuildCache().succeeds("cacheable", "-Prun=2")
-        withBuildCache().succeeds("cacheable", "-Prun=4")
-        withBuildCache().succeeds("cacheable", "-Prun=6")
-        then:
-        def recentlyUsed = originalList.findAll {
-            it.lastModified() > oldTime
-        }
-        recentlyUsed.size() == 3
-
-        when:
-        markCacheDirForCleanup()
-        and:
-        withBuildCache().succeeds("cacheable")
-
-        then:
-        def newList = listCacheFiles()
-        newList.size() == MAX_CACHE_SIZE-1
-        newList.containsAll(recentlyUsed)
+    static def configureCacheEviction() {
+        """
+            buildCache {
+                local {
+                    removeUnusedEntriesAfterDays = ${MAX_CACHE_AGE}
+                }
+            }
+        """
     }
 
-    def "does not cleanup on every build"() {
-        when:
-        runMultiple(MAX_CACHE_SIZE*2)
-        then:
-        def originalList = listCacheFiles()
-        // build cache hasn't been cleaned yet
-        originalList.size() == MAX_CACHE_SIZE*2
-        calculateCacheSize(originalList) >= MAX_CACHE_SIZE
+    def "cleans up entries"() {
+        // Make sure cache directory is initialized
+        run()
+        def lastCleanupCheck = gcFile().makeOlder().lastModified()
 
         when:
-        markCacheDirForCleanup()
-        and:
-        withBuildCache().succeeds("cacheable")
+        def trashFile = cacheDir.file("0" * 32).createFile()
+        trashFile.lastModified = System.currentTimeMillis() - TimeUnit.DAYS.toMillis(MAX_CACHE_AGE) * 2
+        run()
         then:
-        def newList = listCacheFiles()
-        newList.size() == MAX_CACHE_SIZE-1
-        // Cache should be under the target size
-        calculateCacheSize(newList) <= MAX_CACHE_SIZE
-        def lastCleanedTime = gcFile().lastModified()
+        trashFile.assertIsFile()
+        assertCacheWasNotCleanedUpSince(lastCleanupCheck)
 
-        // build cache shouldn't clean up again
         when:
-        runMultiple(MAX_CACHE_SIZE)
+        lastCleanupCheck = markCacheForCleanup()
+        run()
         then:
-        lastCleanedTime == gcFile().lastModified()
+        trashFile.assertDoesNotExist()
```

Comment:
Should we also add a file to the cache which is younger than the `MAX_CACHE_AGE` and does not get deleted? Or are there some other tests for this?